include/seastar/core/metrics_api.hh

```diff
@@ -356,6 +395,8 @@ struct config {
     sstring hostname;
 };
 
+using internalized_set = std::set<internalized_holder, std::less<>>;
```

Comment:
This would be a candidate for unordered_set. But since you need ordered compares for the other container, we might as well keep this as is.